lib/rules/yoda.js

```diff
@@ -209,7 +212,8 @@ module.exports = {
                 return ((tokenBefore = sourceCode.getTokenBefore(node)) &&
                     tokenBefore.value === "(" &&
                     (tokenAfter = sourceCode.getTokenAfter(node)) &&
-                    tokenAfter.value === ")");
+                    tokenAfter.value === ")"
```

Comment:
I'm a bit worried about this side-effect. Having a paren after the statement is valid indentation, and we will probably break peoples' builds if we report it as an issue. We should be sure to fix this before we merge this PR.

(This was also mentioned [here](https://github.com/eslint/eslint/pull/7197#pullrequestreview-1375065). I'm leaving this review to make sure we don't forget about it and merge the PR without fixing this.)